README.md

```diff
@@ -2,18 +2,89 @@
 
 A proposed [WebAssembly System Interface](https://github.com/WebAssembly/WASI) API.
 
-This proposal currently only contains the proposed Wit interfaces with light
-explanation in comments; more work is necessary to fully document the proposal.
-The Wit comments annotate where the proposed interface is expected to change in
-the short term (for Preview2) once resources and handles are re-added to Wit,
-and then after that (for Preview3) once native stream support is added to the
-Component Model and Wit.
-
-The `wit` directory currently validates and can generate bindings with:
+### Current Phase
+
+wasi-http is currently in [Phase 1](https://github.com/WebAssembly/WASI/blob/main/Proposals.md#phase-1---feature-proposal-cg).
+
+### Champions
+
+* Piotr Sikora
+* Jiaxiao Zhou
+* Dan Chiarlone
+* David Justice
+* Luke Wagner
+
+### Phase 4 Advancement Criteria
+
+WASI-http must have at least two complete independent implementations. One
+implementation must execute in a browser and may be implemented in terms of the
+[Fetch API] using JavaScript. The other implementation must be implemented
```

Comment:
Should this say [FetchEvent](https://developer.mozilla.org/en-US/docs/Web/API/FetchEvent) ?

Reply:
Good question.  The Fetch API (specified [here](https://fetch.spec.whatwg.org/)) is for *outgoing* requests, but you're right that FetchEvent is also necessary for *incoming* requests, and I think they're in the Service Worker spec ([here](https://w3c.github.io/ServiceWorker/)), which is separate.  So I guess a fix would be to mention FetchEvent in addition to the Fetch API, linking to the SW spec, if that sounds right to you.

Reply:
Hopefully those browser APIs mirror each other and there is no surprise. Maybe linking both it the right thing to do.

There will be interesting details about streaming behavior of request and response. I know that not all browsers implement streaming in `fetch`. They buffer first until they have full body. Only Chromium knows how to stream. 

And I don't know that is also true for service worker.

That said, maybe this WASI api could consider those details. 

I guess other implementers may have similar limitations.

Also aborting, timeouts and canceling of promises is interesting in this context (in order to safe bandwidth and/or memory when you don't need the data anymore). Browsers implement those gradually ...

Sorry I have not looked at this WIT yet.